src/main/java/uk/gov/hmcts/darts/event/mapper/EventMapper.java

```diff
@@ -3,49 +3,86 @@
 import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Component;
 import uk.gov.hmcts.darts.common.entity.EventEntity;
-import uk.gov.hmcts.darts.event.model.AdminGetEventForIdResponseResult;
+import uk.gov.hmcts.darts.event.model.AdminGetEventResponseDetails;
+import uk.gov.hmcts.darts.event.model.AdminGetVersionsByEventIdResponseResult;
 import uk.gov.hmcts.darts.event.model.CourthouseResponseDetails;
 import uk.gov.hmcts.darts.event.model.CourtroomResponseDetails;
-import uk.gov.hmcts.darts.event.model.EventMappingDetails;
+import uk.gov.hmcts.darts.event.model.EventMapping;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
 
 @Component
 @RequiredArgsConstructor
 public class EventMapper {
 
-    public AdminGetEventForIdResponseResult mapToAdminGetEventsResponseForId(EventEntity eventEntity) {
-        AdminGetEventForIdResponseResult adminGetEventsForIdResponse = new AdminGetEventForIdResponseResult();
+    public AdminGetEventResponseDetails mapToAdminGetEventsResponseForId(EventEntity eventEntity) {
+        AdminGetEventResponseDetails adminGetEventResponseDetail = new AdminGetEventResponseDetails();
 
-        adminGetEventsForIdResponse.setId(eventEntity.getId());
-        adminGetEventsForIdResponse.setDocumentumId(eventEntity.getLegacyObjectId());
-        adminGetEventsForIdResponse.setSourceId(eventEntity.getEventId());
-        adminGetEventsForIdResponse.setMessageId(eventEntity.getMessageId());
-        adminGetEventsForIdResponse.setText(eventEntity.getEventText());
-        adminGetEventsForIdResponse.setIsLogEntry(eventEntity.isLogEntry());
+        adminGetEventResponseDetail.setId(eventEntity.getId());
+        adminGetEventResponseDetail.setDocumentumId(eventEntity.getLegacyObjectId());
```

Comment:
`documentum_id` is not specified in the response in the LLD, should it be?

Reply:
Added to LLD